### ISSUE TITLE
Extend the models

### ISSUE THREAD

**Issue by rblijenbergh** [musonza/chat#275]
Is there a way I can extend the models (like Conversation, Participation etc.)?

**Comment by samerzaki** [comment]
me too asking that question

**Comment by theimerj** [comment]
ideal and cleanest would be to have the possibility of providing custom model classes in config - pretty easy to implement, you just have to go through the codebase.

@musonza would you like this functionality? if yes, i might make a PR when i have a bit more time

**not ideal, may potentially introduce some problems and will not solve every use case**
one way to use own models which extend package models is by doing this

Create your own model
```php
<?php

namespace Domain\Chat\Models;

use Illuminate\Database\Eloquent\Relations\HasMany;
use Musonza\Chat\Models\Conversation as ChatConversation;

class Conversation extends ChatConversation
{
    /**
     * Messages in conversation.
     *
     * @return HasMany
     */
    public function messages(): HasMany
    {
        return $this->hasMany(Message::class, 'conversation_id'); //->with('sender');
    }
}
```

Create a trait which points to your own model implementation
```php
<?php

namespace Domain\Chat\Traits;

use Domain\Chat\Models\Conversation;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

trait InteractsWithChat
{
    /**
     * Chat relation.
     *
     * @return BelongsTo
     */
    public function chat(): BelongsTo
    {
        return $this->belongsTo(Conversation::class, 'conversation_id');
    }
}
```
this is just a quick thought, i havent thought much about the consequences, but i needed to add one relation of my own to `Message` model, so i will see how it goes for now

**Comment by Forsakenrox** [comment]
Yep, extending package models - core functionality on my opinion. But solution with traits - dirty. Maybe extending of base Chat:class will be is more flexible

**Comment by tommie24** [comment]
Best way is to extend the core model and add your own functions like @theimerj suggested.

**Comment by musonza** [comment]
Closing this issue due to inactivity. Please refer to the current documentation and README for usage guidance. If you still have questions or are experiencing issues with the latest version, feel free to open a new issue.
